worksheet_D/TicTacToe.py

```diff
@@ -0,0 +1,114 @@
+class OxoBoard:
+
+    def __init__(self,x,y):
+        self.x = x
+        self.y = y
+        self.board = {(0,0):0,(0,1):0,(0,2):0,(1,0):0,(2,0):0,(1,1):0,(2,2):0,(1,2):0,(2,1):0}
+
+    def get_square(self, x, y):
+        return self.board[x,y]
+
+    def set_square(self, x, y, full):
+        if self.board[x,y] == 0:
+            self.board[x,y] = full
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        spaces = 9
+        for i in xrange(0,3):
+            for j in xrange(0,3):
+                if self.board[i, j] > 0:
+                    spaces -= 1
+                if spaces == 0:
+                    return True
+        return False
+
+    def get_winner(self, player_number):
+    #See if there is a winner by checking diagonals from bottom to top
+         if self.board[1, 1] != 0:
+             if self.board[1, 1] == self.board[0, 0]:
+                 if self.board[2, 2] == self.board[1, 1]:

```

Comment:
It would be neater to combine these conditions into a single `if` using `and`. Or even better, use Python's ability to chain `==` and `!=` to check more than two values.